packages/jest-cli/src/TestRunner.js

```diff
@@ -396,6 +400,57 @@ class TestRunner {
     if (config.notify) {
       this.addReporter(new NotifyReporter(this._startRun));
     }
+
+    return undefined;
+  }
+
+  /**
+   * Add Custom reporters to Jest
+   * Custom reporters can be added to Jest using the reporters option in Jest
+   * Config. The format for adding a custom reporter is following:
+   *
+   * Format for the custom reporters
+   *
+   * "reporters": [
+   *    ["reporterPath/packageName", { option1: 'fasklj' }], 
+   *    // Format if we want to specify options
+   *    
+   *    "reporterName"
+   *    // Format if we don't want to specify any options
+   * ]
+   */
+  _addCustomReporters(reporters: any) {
+    let reporterPath, reporterConfig;
+
+    reporters.forEach(entry => {
+      if (Array.isArray(entry)) {
+        [reporterPath, reporterConfig] = entry;
+      } else {
+        if (typeof entry !== 'string') {
+          throw new Error(`
+            Unexpected custom reporter configuration.
+            Expected to get either a path string or an array of [path, confg]
+            got: ${JSON.stringify(entry)}
+          `);
```

Comment:
@thymikee @dmitriiabramov  Ideally will this be added to `jest-validate` or I guess Validate can be used here, since it's kind of a validation error.